python/pyspark/sql/session.py

```diff
@@ -214,8 +214,14 @@ def __init__(self, sparkContext, jsparkSession=None):
         self._wrapped = SQLContext(self._sc, self, self._jwrapped)
         _monkey_patch_RDD(self)
         install_exception_handler()
-        if SparkSession._instantiatedContext is None:
-            SparkSession._instantiatedContext = self
+        if SparkSession._instantiatedSession is None:
+            SparkSession._instantiatedSession = self
+        else:
+            # If we had an instantiated SparkSession attached with a SparkContext
+            # which is stopped now, we need to renew the instantiated SparkSession.
+            # Otherwise, we will use invalid SparkSession when we call Builder.getOrCreate.
+            if SparkSession._instantiatedSession._sc._jsc is None:
+                SparkSession._instantiatedSession = self
```

Comment:
`SparkSession._instantiatedSession._sc` is from the `sparkContext` passed into `__init__`. I think it should not be None.